pkg/providers/launchtemplate/suite_test.go

```diff
@@ -1743,8 +1967,36 @@ func ExpectLaunchTemplatesCreatedWithUserData(expected string) {
 		userData, err := base64.StdEncoding.DecodeString(*input.LaunchTemplateData.UserData)
 		ExpectWithOffset(2, err).To(BeNil())
 		// Newlines are always added for missing TOML fields, so strip them out before comparisons.
+		fmt.Print(string(userData))
 		actualUserData := strings.Replace(string(userData), "\n", "", -1)
 		expectedUserData := strings.Replace(expected, "\n", "", -1)
 		ExpectWithOffset(2, actualUserData).To(Equal(expectedUserData))
 	})
 }
+
+func ExpectLaunchTemplatesCreatedWithUserDataF(fn func(userData string)) {
```

Comment:
Callbacks tend to be an anti-pattern unless there is a good reason to use them. I'm personally not convinced that there's a good reason that this function needs to exist. Similar to another comment that I made before, I spent. more time trying to figure out what the function did rather than just being able to reason about it from the function name. That's a good indication to me that the function might not be useful on its own. Consider just calling these things inline in the tests. 

As an aside: I do really like the function on L1987 and think that it's valuable